platform/core/src/utils/guid.js

```diff
@@ -1,13 +1,16 @@
 /**
- * Create a random GUID
+ * Create a random GUID using Web Crypto API for better randomness
  *
  * @return {string}
  */
 const guid = () => {
   const getFourRandomValues = () => {
-    return Math.floor((1 + Math.random()) * 0x10000)
-      .toString(16)
-      .substring(1);
+    // Generate 2 random bytes using Web Crypto API
+    const array = new Uint8Array(2);
+    crypto.getRandomValues(array);
```

Comment:
You have to leave the old one in place for non-secure environments as the crypto isn't available in some environments.

Reply:
Added back.